ni_measurementlink_service/_internal/service_manager.py

```diff
@@ -85,24 +87,31 @@ def start(
                 ("grpc.max_send_message_length", -1),
             ],
         )
-        servicer_v1 = MeasurementServiceServicerV1(
-            measurement_info,
-            configuration_parameter_list,
-            output_parameter_list,
-            measure_function,
-        )
-        v1_measurement_service_pb2_grpc.add_MeasurementServiceServicer_to_server(
-            servicer_v1, self.server
-        )
-        servicer_v2 = MeasurementServiceServicerV2(
-            measurement_info,
-            configuration_parameter_list,
-            output_parameter_list,
-            measure_function,
-        )
-        v2_measurement_service_pb2_grpc.add_MeasurementServiceServicer_to_server(
-            servicer_v2, self.server
-        )
+        for interface in service_info.provided_interfaces:
+            if interface == _V1_INTERFACE:
+                servicer_v1 = MeasurementServiceServicerV1(
+                    measurement_info,
+                    configuration_parameter_list,
+                    output_parameter_list,
+                    measure_function,
+                )
+                v1_measurement_service_pb2_grpc.add_MeasurementServiceServicer_to_server(
+                    servicer_v1, self.server
+                )
+            elif interface == _V2_INTERFACE:
+                servicer_v2 = MeasurementServiceServicerV2(
+                    measurement_info,
+                    configuration_parameter_list,
+                    output_parameter_list,
+                    measure_function,
+                )
+                v2_measurement_service_pb2_grpc.add_MeasurementServiceServicer_to_server(
+                    servicer_v2, self.server
+                )
+            else:
+                raise Exception(
```

Comment:
Please don't raise the base `Exception` type. Use a more specific type. I think `ValueError` or `RuntimeError` would be ok here.